javalanglib/src/main/scala/java/lang/Enum.scala

```diff
@@ -0,0 +1,22 @@
+package java.lang
+
+abstract class Enum[E <: Enum[E]] protected(val name: String, val ordinal: Int)
```

Comment:
The parameters should be non-val `_name` and `_ordinal`, and you should have separate definitions of `final def name()` and `final def ordinal()`. This is because https://github.com/scala-js/scala-js/blob/master/CODINGSTYLE.md#parameterless-methods

Also, there should be a space between `protected` and `(`